### ISSUE TITLE
Error when using string_to_mv() from a subquery

### ISSUE THREAD

**Issue by phillbrown** [apache/druid#14006]
### Affected Version

25.0

### Description

Steps to reproduce:

1. Load the [attached](https://github.com/apache/druid/files/11123016/string-to-mv.zip) data using native batch ingestion with the [attached](https://github.com/apache/druid/files/11123016/string-to-mv.zip) spec
2. Run the following query using the SQL native engine

```
SELECT STRING_TO_MV(mvstring, ',')
FROM (
  SELECT DISTINCT "parent_id", MV_TO_STRING("multi_values", ',') AS mvstring
  FROM "example"
  GROUP BY 1, 2
)
```

The query should return the following error:

```
Error: Unknown exception

Cannot coerce [[Ljava.lang.String;] to VARCHAR

org.apache.druid.java.util.common.ISE
```

Interestingly the query runs as expected if we use the ms query engine. I'm using the out-the-box auto configuration.

**Comment by LakshSingla** [comment]
Thanks for reporting the issue along with the steps to repro it. Raised a PR to solve the same.

**Comment by LakshSingla** [comment]
Resolved by #14260
